git-tar/function/ops.go

```diff
@@ -489,3 +487,20 @@ func importSecrets(pushEvent sdk.PushEvent, stack *stack.Services, clonePath str
 
 	return nil
 }
+
+func formatTemplateRepos() []string {
+	templateRepos := []string{"https://github.com/openfaas/templates"}
+
+	if envTemplates := os.Getenv("custom_templates"); len(envTemplates) > 0 {
+		customTemplates := strings.Split(envTemplates, ",")
+		for _, repo := range customTemplates {
+			if strings.Contains(envTemplates, "https://") || strings.Contains(envTemplates, "http://") {
```

Comment:
This does not count as URL validation. You should be using https://godoc.org/net/url#URL.Parse from the Go stdlib.

I also can't see any error handling here. `func formatTemplateRepos() []string` should be: `func formatTemplateRepos() ([]string, error)`

Fix this up in the linked issue #319